File: check_quotes.ts
import fs from 'fs';

const content = fs.readFileSync('supabase/functions/uazapi-webhook/index.ts', 'utf8');
let inSingleQuote = false;
let inDoubleQuote = false;
let inBacktick = false;
let escaped = false;

const lines = content.split('\n');
for (let i = 0; i < lines.length; i++) {
  const line = lines[i];
  for (let j = 0; j < line.length; j++) {
    const char = line[j];
    if (escaped) {
      escaped = false;
      continue;
    }
    if (char === '\\') {
      escaped = true;
      continue;
    }
    if (inSingleQuote) {
      if (char === "'") inSingleQuote = false;
      continue;
    }
    if (inDoubleQuote) {
      if (char === '"') inDoubleQuote = false;
      continue;
    }
    if (inBacktick) {
      if (char === '`') inBacktick = false;
      continue;
    }
    if (char === "'") inSingleQuote = true;
    else if (char === '"') inDoubleQuote = true;
    else if (char === '`') inBacktick = true;
  }
}

if (inSingleQuote) console.log('Unclosed single quote');
if (inDoubleQuote) console.log('Unclosed double quote');
if (inBacktick) console.log('Unclosed backtick');

File: check_quotes.py
with open('supabase/functions/uazapi-webhook/index.ts', 'r') as f:
    content = f.read()

in_s = None # ' " `
in_c = None # // /*
esc = False

i = 0
while i < len(content):
    c = content[i]
    
    if in_c == '//':
        if c == '\n':
            in_c = None
        i += 1
        continue
    if in_c == '/*':
        if content[i:i+2] == '*/':
            in_c = None
            i += 2
        else:
            i += 1
        continue
        
    if in_s:
        if esc:
            esc = False
        elif c == '\\':
            esc = True
        elif c == in_s:
            in_s = None
        i += 1
        continue
    
    if content[i:i+2] == '//':
        in_c = '//'
        i += 2
        continue
    if content[i:i+2] == '/*':
        in_c = '/*'
        i += 2
        continue
        
    if c in ["'", '"', '`']:
        in_s = c
        start_line = content.count('\n', 0, i) + 1
        i += 1
        continue
        
    i += 1

if in_s:
    print(f"Unclosed {in_s} starting at line {start_line}")
else:
    print("All quotes closed")
